starport/pkg/cosmosanalysis/app/app.go

```diff
@@ -69,3 +70,161 @@ func CheckKeeper(path, keeperName string) error {
 	}
 	return nil
 }
+
+// FindRegisteredModules looks for all the registered modules in the App
+// It connects the imported go module to the registered module and returns a list of the go modules that are registered
+// It does so by:
+// 1. Mapping out all the imports and named imports
+// 2. Looking for the call to module.NewBasicManager and finds the modules registered there
+// 3. Looking for the implementation of RegisterAPIRoutes and find the modules that call their RegisterGRPCGatewayRoutes
+func FindRegisteredModules(pathToApp string) ([]string, error) {
+	fileSet := token.NewFileSet()
+	pkgs, err := parser.ParseDir(fileSet, pathToApp, nil, 0)
+	if err != nil {
+		return []string{}, err
+	}
+
+	var basicModules []string
+	for _, pkg := range pkgs {
+		for p, f := range pkg.Files {
```

Comment:
There is an error from the integration tests about `findBasicManagerModule`. To fix, maybe instead of checking all files, we can discover the path to `app/app.go` and only operate in this file?

Reply:
I wasn't able to reproduce the integration test failure locally, but I extended the unit test to find it. The loop now ignore any non app.go file.